### PR TITLE
Replaces Stripe stubs with the real account IDs

### LINKED ISSUE #11949: [Integration/API tests] app/models/stripe_account.rb
Follow-up from https://github.com/openfoodfoundation/openfoodnetwork/pull/11900#issuecomment-1841892371.

### PR DESCRIPTION
**Pull request by filipefurtad0**
#### What? Why?

Related to Epic #11890.
Closes #11949.

Sets up the spec with Stripe credentials, so our code can actually make the real calls to Stripe; Records responses.

Also adds STRIPE_ACCOUNT as sensitive data to VCR setup.

#### What should we test?

- Green build.

#### Release notes

Changelog Category (reviewers may add a label for the release notes):

- [ ] User facing changes
- [ ] API changes (V0, V1, DFC or Webhook)
- [x] Technical changes only
- [ ] Feature toggled

The title of the pull request will be included in the release notes.

### CODE REVIEW

File: spec/models/stripe_account_spec.rb [hunk 1]
```diff
@@ -4,28 +4,30 @@
...
     end
 
     context "when the Stripe API disconnect fails" do
-      before do
-        stub_request(:post, "https://connect.stripe.com/oauth/deauthorize").
-          with(body: { "client_id" => client_id, "stripe_user_id" => stripe_user_id }).
-          to_return(status: 400, body: JSON.generate(error: 'invalid_grant',
-                                                     error_description: "Some Message"))
-      end
-
```
Comment (filipefurtad0) [inline spec/models/stripe_account_spec.rb:28]: I wonder if this stub was correct in the first place, or rather the test setup is not appropriate. From the real Stripe call, we get the response:

- with a `status: 200`
- with  
```
body:
      encoding: UTF-8
      string: |-
        {
          "error": "invalid_client",
          "error_description": "No such application: 'cus_PDCEEWCCEIDICb'"
        }
```

A status 400 would be a bad request?...

File: spec/models/stripe_account_spec.rb [hunk 2]
```diff
@@ -24,18 +24,28 @@
...
+          StripeAccount.where(stripe_user_id:), :count
+        ).from(1).to(0)
       end
     end
 
     context "when the Stripe API disconnect succeeds" do
       before { Stripe.client_id = client_id }
 
       it "destroys the record" do
+        # returns status 200
```
[the remark this answers is not in the document]
Comment (filipefurtad0) [reply spec/models/stripe_account_spec.rb:41]: The issue here is that apparently, stripe_account is not connected:

```
        {
          "error": "invalid_client",
          "error_description": "This application is not connected to stripe account <HIDDEN_ACCOUNT>, or that account does not exist."
        }
```

So, it does no pass the disconnect test. This is the case, even if use `create_account` from `lib/stripe/account_connector.rb`... Maybe I should test this file before proceeding here.

[the remark this answers is not in the document]
Comment (filipefurtad0) [reply spec/models/stripe_account_spec.rb:52]: Ok, got it now - we can actually create a simple account with a direct API call to Stripe, and then pass it as an argument to our `deauthorize_and_destroy` [method defined here](https://github.com/openfoodfoundation/openfoodnetwork/blob/589ae4db4412983a5e4e7372fad14978f7e75fa5/app/models/stripe_account.rb#L10).

If we do this, then the account can be disconnected and Bugsnag is not triggered.

File: spec/models/stripe_account_spec.rb [hunk 3]
```diff
@@ -24,18 +24,28 @@
...
       it "destroys the record and notifies Bugsnag" do
+        # returns status 401
         expect(Bugsnag).to receive(:notify)
-        stripe_account.deauthorize_and_destroy
-        expect(StripeAccount.all).to_not include(stripe_account)
+        expect {
+          stripe_account.deauthorize_and_destroy
+        }.to change(
+          StripeAccount.where(stripe_user_id:), :count
+        ).from(1).to(0)
```
Comment (mkllnk) [inline spec/models/stripe_account_spec.rb:33]: Excellent!

**Review by mkllnk — APPROVED** [review]
Thank you!

**Pull request merged**